File: windows/tests/ConnectorControl.Core.Tests/CredentialHeuristicsTests.cs
namespace ConnectorControl.Core.Tests;

/// <summary>Mirror: Tests/ConnectorControlCoreTests/CredentialHeuristicsTests.swift</summary>
public class CredentialHeuristicsTests
{
    [Fact]
    public void KnownPrefixesAndLongMixedTokens()
    {
        Assert.True(CredentialHeuristics.LooksLikeCredential("sk-live-9f3a"));
        Assert.True(CredentialHeuristics.LooksLikeCredential("ghp_abc123"));
        Assert.True(CredentialHeuristics.LooksLikeCredential("xoxb-1-2"));
        Assert.True(CredentialHeuristics.LooksLikeCredential("Bearer abc"));
        Assert.True(CredentialHeuristics.LooksLikeCredential("a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6"));
        Assert.True(CredentialHeuristics.LooksLikeCredential("a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4"));   // 32 characters is the boundary
        Assert.False(CredentialHeuristics.LooksLikeCredential("a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d"));   // 31 is below it
    }

    [Fact]
    public void OrdinaryArgumentsAreNotFlagged()
    {
        Assert.False(CredentialHeuristics.LooksLikeCredential("-y"));
        Assert.False(CredentialHeuristics.LooksLikeCredential("@dbt/mcp"));
        Assert.False(CredentialHeuristics.LooksLikeCredential("https://example.com/mcp"));
        Assert.False(CredentialHeuristics.LooksLikeCredential("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"));
        Assert.False(CredentialHeuristics.LooksLikeCredential("${CC_NEEDS:token}"));
    }
}

File: Tests/ConnectorControlCoreTests/CredentialHeuristicsTests.swift
import XCTest
@testable import ConnectorControlCore

/// Mirror: windows/tests/ConnectorControl.Core.Tests/CredentialHeuristicsTests.cs
final class CredentialHeuristicsTests: XCTestCase {
    func testKnownPrefixesAndLongMixedTokens() {
        XCTAssertTrue(CredentialHeuristics.looksLikeCredential("sk-live-9f3a"))
        XCTAssertTrue(CredentialHeuristics.looksLikeCredential("ghp_abc123"))
        XCTAssertTrue(CredentialHeuristics.looksLikeCredential("xoxb-1-2"))
        XCTAssertTrue(CredentialHeuristics.looksLikeCredential("Bearer abc"))
        XCTAssertTrue(CredentialHeuristics.looksLikeCredential("a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6"))
        XCTAssertTrue(CredentialHeuristics.looksLikeCredential("a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4"), "32 characters is the boundary")
        XCTAssertFalse(CredentialHeuristics.looksLikeCredential("a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d"), "31 characters is below it")
    }
    func testOrdinaryArgumentsAreNotFlagged() {
        XCTAssertFalse(CredentialHeuristics.looksLikeCredential("-y"))
        XCTAssertFalse(CredentialHeuristics.looksLikeCredential("@dbt/mcp"))
        XCTAssertFalse(CredentialHeuristics.looksLikeCredential("https://example.com/mcp"))
        XCTAssertFalse(CredentialHeuristics.looksLikeCredential("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"))
        XCTAssertFalse(CredentialHeuristics.looksLikeCredential("${CC_NEEDS:token}"))
        // A digit is a decimal digit: a superscript or a vulgar fraction is not one.
        XCTAssertFalse(CredentialHeuristics.looksLikeCredential("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\u{B2}\u{BD}"))
    }
}
